### DISCUSSION TITLE
MultiSpace vs DisjointPDF

### DISCUSSION THREAD

**Discussion by jonas-eschle** [zfit/zfit-development#86] [Ideas]
MultiSpace, i.e. multiple, disjoint spaces are currently part of zfit. They offer a convenient way to express disjoint limits for normalization, plotting, sampling, integration, but they also add a multiplicative complication to it: everything that takes a space will need to handle the case of a multispace.

And that can be hard, i.e. see the breaking KDE preprocessing of the data for boundary conditions.

Alternatively, we could have a DisjointPDF that is maybe masked in the middle or holds arbitrarily many spaces for the normalization (if needed). The former changes the PDF, the latter changes the normalization.

Thoughts?

**Comment by schmitse** [comment]
Hi, :) 

Very interesting topic.

I think a disjoint KDE adds more complexity regardless of the way that spaces are treated, since i think you cannot get past having to be able to appropriately mirror/preprocess input data over a range of disjoint spaces. This would need to be implemented then regardless of the way one treats the underlying space. 

Consider the example where there is no data in an enclosed region of the space and you only mirror on the outer observable edges, you will always have boundary effects inside the core of your space from missing to mirror the KDE there. 

For analytic PDFs i think that all three given options (MultiSpace, masks, multiple spaces for norm) are viable but im not sure where complications might arise in the latter two. 

Cheers :smiley:
